### PR TITLE
Lesson 1.5

### PR DESCRIPTION
**Pull request by viniciuspietscher**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
-  const [todoList, setTodoList] = useState([])
+  const [todoList, setTodoList] = useSemiPersistentState()
 
   const addTodo = (newTodo) => {
     setTodoList([...todoList, newTodo])
   }
 
   return (
-    <div>
+    <>
```
Comment (kwilson05) [inline src/App.js:27]: Fragments!! 💯

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,37 @@
-import { useState } from "react"
+import { useState, useEffect } from "react"
 import TodoList from "./TodoList"
 import AddTodoForm from "./AddTodoForm"
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList"))
+      ? JSON.parse(localStorage.getItem("savedTodoList"))
+      : []
```
Comment (kwilson05) [inline src/App.js:9]: good catch, on setting the initial state! 🏅 

You can also set the initial state like this:
`useState(JSON.parse(localStorage.getItem("savedTodoList")) || [])`

^ This is saying try to run the `JSON.parse(localStorage.getItem("savedTodoList"))`, if no value returned or `undefined`. run `[]`.

**Review by kwilson05 — APPROVED** [review]
Good work, Vini! I only made one minor comment 🏅

File: src/App.js (hunk 2 above)
Comment (viniciuspietscher) [reply src/App.js:9]: Thank you!

**Pull request merged**